src/hoc/BadgerBase/BadgerBase.js

```diff
@@ -283,6 +308,56 @@ const BadgerBase = (Wrapped: React.AbstractComponent<any>) => {
 			this.setState({ intervalUnconfirmed: intervalUnconfirmedNext });
 		};
 
+		setupWatchInvoice = async () => {
+			const { paymentRequestUrl } = this.props;
+
+			this.ws = new WebSocket(
```

Comment:
Let's put the reference to `ws` in the `state` for consistency with the rest of the project, then set it up in the `componentDidMount` hook and cleanup in the `componentWillUnmount` hook

Reply:
2875f6a